luigi/server.py

```diff
@@ -73,6 +92,62 @@ def get_template_path(self):
         return pkg_resources.resource_filename(__name__, 'templates')
 
 
+class AllRunHandler(BaseTaskHistoryHandler):
+
+    def get(self):
+        # get task history db in the same path
+        engine = create_engine('sqlite:///.task_history.db', echo=True)
+        conn_sqlite = engine.connect()
+        query = "select id, name from tasks order by id desc"
+        task_list = conn_sqlite.execute(query)
+        tasknames = []
+        for task in task_list:
+            tasknames.append(task['name'])
+        self.render("menu.html", tasknames=tasknames)
+
+
+class SelectedRunHandler(BaseTaskHistoryHandler):
+
+    def get(self, name):
+        tasks = {}
+        chartResults = {}
+        tableResults = []
+        # get task history db in the same path
+        engine = create_engine('sqlite:///.task_history.db', echo=True)
+        conn_sqlite = engine.connect()
+        task_query = "select id, name from tasks"
+        history_query = "select id, task_id, event_name, ts from task_events"
+        task_list = conn_sqlite.execute(task_query)
+        task_history = conn_sqlite.execute(history_query)
+        for task in task_list:
+            task_seq = task['id']
+            task_name = task['name']
+            tasks[task_seq] = str(task_name)
+        for task in task_history:
+            if tasks.get(task['task_id']) == str(name):
+                status = str(task['event_name'])
+                if status not in chartResults:
+                    chartResults[status] = []
+                chartResults[status].append(({
+                    'id': str(task['id']), 'task_id': str(task['task_id']),
+                    'x': from_utc(str(task['ts'])), 'y': 0, 'next_process': ''}))
+                tableResults.append({
+                    'id': str(task['id']), 'taskName': str(name), 'task_id': str(task['task_id']),
+                    'status': str(task['event_name']), 'datetime': str(task['ts']),
+                    'timestamp': from_utc(str(task['ts'])), 'processing_time': 0})
+        chartResults = json.dumps(chartResults)
+        tableResults = json.dumps(tableResults)
+        chartResults = tornado.escape.xhtml_unescape(str(chartResults))
+        self.render('history.html', name=name, chartResults=chartResults, tableResults=tableResults)
+
+
+def from_utc(utcTime, fmt="%Y-%m-%d %H:%M:%S.%f"):
+    """convert UTC time string to time.struct_time: change datetime.datetime to time, return time.struct_time type"""
+    time_struct = datetime.datetime.strptime(utcTime, fmt)
+    date = int(time.mktime(time_struct.timetuple())) * 1000
```

Comment:
Minor point but the millisecond standard is a Java/Javascript convention that doesn't exist in other language. I would prefer if the millisecond conversion takes place in the frontend instead.